app/models/manageiq/providers/openstack/cloud_manager/cloud_volume/operations.rb

```diff
@@ -8,6 +8,7 @@ def validate_detach_volume
   end
 
   def raw_attach_volume(server_ems_ref, device = nil)
+    device = nil if (device && device.empty?)
```

Comment:
Minor Style: Please remove the parens

Additionally, you can do the following... `device = nil if device.try(:empty?)`  @Ladas @blomquisg Not sure which style you prefer there.